### ISSUE TITLE
Updating extensions in extra directories causes copy to be downloaded

### ISSUE THREAD

**Issue by code-asher** [cdr/code-server#1157]
It should update the existing extension in the extra directory instead of downloading a new copy into the main directory.

**Comment by nhooyr** [comment]
@code-asher Is this still an issue?

**Comment by code-asher** [comment]
Yup, confirmed on 3.5.0. The extension will still work but constantly says "reload required" which makes me think it's getting into a strange state by loading the extension twice.

And if you uninstall it the extension won't be listed in the extension list anymore yet will still be activated. Seems to have something to do with the `.obsolete` file they generate instead of actually removing extensions from disk.

**Comment by code-asher** [comment]
It looks like updating is supposed to download a new copy and leave the old one intact though so I'll rewrite the issue a bit to make it more clear as to what the actual problems are.

**Comment by jabbera** [comment]
I just ran into this today in my JH installation.

**Comment by neupsh** [comment]
I believe I am also running into this issue. I have different sets of extensions enabled in different workspaces, but I have all of my installed extensions enabled at once and the ones that were supposed to be disabled for the current workspace show up as "Reload Required" (to disable this extension). Reloading does not help as it goes back to the same state.
